lib/fizzy/instantiate.hpp

```diff
@@ -21,8 +21,31 @@ namespace fizzy
 struct ExecutionResult;
 struct Instance;
 
+class ThreadContext
+{
+public:
+    int depth = 0;
+
+    void acquire() noexcept { ++depth; }
```

Comment:
Do these need to be public? Are these supposed to be used directly or only via the Guard?